vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java

```diff
@@ -94,12 +94,10 @@ public AbstractNumberField(SerializableFunction<String, T> parser,
         setValueChangeMode(ValueChangeMode.ON_CHANGE);
 
         addValueChangeListener(e -> validate());
-        addInvalidChangeListener(e -> {
-            // If invalid is updated from client to false, check it
-            if (e.isFromClient() && !e.isInvalid()) {
-                validate();
-            }
-        });
+
+        addAttachListener(e -> getElement()
+                .executeJs("$0.validate = function () {return this.checkValidity();}",
+                        getElement()));
```

Comment:
Maybe this should be extracted as a static method somewhere? Maybe in a helper class, eg.
```java
TextFieldValidationUtil.disableClientValidation(Component field)`
```
(the class should be package protected in this case)